cli/serve/serve.go

```diff
@@ -334,6 +334,31 @@ func serverMain(args []string, c cli.Config) error {
 				http.Error(w, "Invalid CN", http.StatusForbidden)
 			})
 		}
+
+		if conf.MutualTLSSANRegex != "" {
+			log.Debugf(`Requiring any SAN matches regex "%s" for client connections`, conf.MutualTLSSANRegex)
+			re, err := regexp.Compile(conf.MutualTLSSANRegex)
+			if err != nil {
+				return fmt.Errorf("malformed SAN regex: %s", err)
+			}
+			server.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if r != nil && r.TLS != nil && len(r.TLS.PeerCertificates) > 0 {
+					for _, name := range r.TLS.PeerCertificates[0].DNSNames {
```

Comment:
The SAN list includes more than DNS names; if you look at the [`Certificate` structure](https://golang.org/pkg/crypto/x509/#Certificate), it has

```
        // Subject Alternate Name values
        DNSNames       []string
        EmailAddresses []string
        IPAddresses    []net.IP
        URIs           []*url.URL
```

so this should probably also check more than the DNS names.

Reply:
Hi Kisom,

I'd considered this, but, should it?

I wasn't too sure what the use-case would be for validating based on those other attributes, but, validating based on the DNSNames is ultimately arbitrary.

If I updated to validate against all of them, do you consider this merge ready/worthy?

-Adam

Reply:
@AdamJacobMuller Other than that, it looks fine.